src/util/PriorityQueue.hxx

```diff
@@ -0,0 +1,62 @@
+#ifndef PRIORITY_QUEUE_HXX_
+#define PRIORITY_QUEUE_HXX_
+
+#include <vector>
+
+/**
+  * @brief Priority queue with removing functionality.
+  */
+template <typename T> class PriorityQueue
+{
+  std::vector<T> m_container;
+
+public:
+  using Iterator = typename std::vector<T>::iterator;
+
+  /**
+    * @return Iterator to beginning of queue.
+    */
+  Iterator begin() { return m_container.begin(); }
+
+  /**
+    * @return Iterator to end of queue.
+    */
+  Iterator end() { return m_container.end(); }
+
+  /**
+    * @brief Check whether queue is empty.
+    * @return true if queue is empty otherwise false.
+    */
+  bool isEmpty() { return m_container.empty(); }
+
+  /**
+    * @brief Get top element from the queue.
+    * @return Top element from queue.
+    */
+  T &top() { return m_container.front(); }
+
+  /**
+    * @brief Add new element to the queue.
+    */
+  void add(T &&element);
+
+  /**
+    * @brief Remove first element from the queue (from top).
+    */
+  void pop();
+
+  /**
+    * @brief Remove task from queue.
+    * @param position Iterator of element to be removed.
+    */
+  void remove(Iterator position);
```

Comment:
This should be called `::erase` to follow STL conventions & also you should overload it to support `const_iterator`s
```suggestion
  void erase(iterator pos);
  void erase(const_iterator pos);
```

Reply:
But I think you should remove completetly in favor of erase_if

Reply:
erase_if is implemented